docs/architecture/self-hosted/authentication/databricks-oauth.md

```diff
@@ -0,0 +1,110 @@
+
+The authentication flow then uses Prophecy-hosted callback URL to capture and process authorization codes, issuing and storing access tokens.
+
+### Token storage
+
+Prophecy stores the refresh token, which are used to renew the refresh token itself, and also get a new access token to maintain authenticated connectivity to Databricks.
+
+The tokens are stored securely, with access limited to authorized Prophecy processes. This includes encrypting tokens before storing them in our database, in the same way that we encrypt other credentials that Prophecy stores.
+
+### Pipeline and Job configuration
+
+As mentioned previously, members of your team developing Pipelines and Jobs can leverage their own personal identity via OAuth to gain access to all Databricks resources from within Prophecy. That means that whatever permissions they have within Databricks (including permissions governed by Unity Catalog) will be enforced in Prophecy as well.
+
+You will see a login overlay at the following points in Prophecy where Databricks API interactions are required:
+
+- Pipeline IDE - Select Fabric
+- Jobs IDE - Select Fabric
+- Jobs IDE - Setting page, in order to fetch list of users and Service Principals
+
+<img
+src={require("./img/data-bricks-oauth-select-fab.png").default}
+alt="Select a Fabric"
+width="70%"
+/>
+
+To proceed through the login overlays, complete the following steps:
+
+1. Click **Continue**. A separate browser tab opens, and Databricks redirects you to the IdP registered in Databricks.
+
+2. Log in with the IdP (or directly with Databricks if there is no IdP). The tab closes and you can proceed with activities such as Pipeline test execution.
+
+:::note
+
+You only need to perform this authentication periodically, depending on the OAuth timeout settings within Databricks. Your Databricks Account Administrator can adjust the timeout setting.
+
+:::
+
+### Project Release and Deployment
+
+Your Prophecy Team Admin is the only member of your team able to perform deployment of the Pipelines and jobs created by the team. Prophecy uses a Databricks Service Principal via Databricks OAuth, to perform this task on behalf of the Team Admin. The Team Admin is the only user who can use this Service Principal, and only for the purpose of deploying team projects.
+
+## OAuth Setup
+
+The Databricks OAuth setup must be completed by both your Databricks Account Admin and your Prophecy Team Admin.
+
+### Databricks Account Admin tasks
+
+Your Databricks Account Admin must complete a one-time procedure on the Databricks side to register Prophecy as an OAuth App available in your account. A registration is required for each private SaaS (on-prem) customer Databricks account.
+
+As the Databricks Account Admin, complete the following steps:
+
+1. On Databricks, navigate to **Account Settings > App connections**.
```

Comment:
@ashvjit-prophecy I don't have access, so if you think it would help, could you provide screenshots of the Databricks App connections?

Reply:
@alexanderahn Sure. Just let me know exactly which screenshots are needed